crates/astria-sequencer/src/proposal/commitment.rs

```diff
@@ -0,0 +1,81 @@
+use std::collections::BTreeMap;
+
+use anyhow::{
+    Context as _,
+    Error,
+    Result,
+};
+use bytes::Bytes;
+use tendermint::merkle::simple_hash_from_byte_vectors;
+
+use crate::transaction::Signed;
+
+/// Called when we receive a `PrepareProposal` or `ProcessProposal` consensus message.
+///
+/// In the case of `PrepareProposal`, we use this function to generate the `commitment_tx`
+/// to be placed at the start of the block.
+///
+/// In the case of `ProcessProposal`, we use this function to generate and verify the
+/// `commitment_tx` expected at the start of the block.
+///
+/// This function sorts the block's `secondary::Action`s contained within the transactions
```

Comment:
The use of `secondary` is confusing because we don't use it anywhere else. Did you mean `sequence::Action`?

Reply:
Yeah in metro they were called secondary transactions, i think naming just got confused here.